velox/exec/Operator.h

```diff
@@ -459,6 +459,17 @@ class Operator : public BaseRuntimeStatWriter {
     return identityProjections_;
   }
 
+  /// Returns a list of filtered projections, which are used for filter pushdown
```

Comment:
> filtered projections

I find this term confusing. Can we find a better name or provide a clearer explanation? Specifically, can you clarify the difference between identityProjection and filteredProjection.

The definition of identity projection "columns that are projected as-is possibly after applying a filter" seems to fit group-by keys. These are returned as is for a subset of rows. Any particular reason HashAgg cannot report group by keys as identity projection? If so, let's update the comment to explain why group-by keys do not qualify.

Reply:
I don't know the particular reason HashAgg cannot report group by keys as identity projection. I just found that in HashAggregation::initialize(), the identityProjections_ is only initialized for the distinct case, as shown below.

```
  if (isDistinct_) {
    for (auto i = 0; i < hashers.size(); ++i) {
      identityProjections_.emplace_back(hashers[i]->channel(), i);
    }
  }
```

Reply:
> I don't know the particular reason HashAgg cannot report group by keys as identity projection. 

Let's figure out whether this is just an omission or there is a good reason.

Reply:
From the code, identityProjections_ is only used `fillOutput()` and `identityProjections()`. In HashAggregation::getOutput(), fillOutput() is called only when isDistinct_ is true. Therefore, for the initialization of identityProjections_ within HashAgg, we can remove the restriction of isDistinct_ as it does not seem to have any side effects. What do you think?

Reply:
This analysis doesn't seem complete because identityProjections() is also used in Driver::pushdownFilters, hence, marking group-by keys as identity projections allows the filter pushdown to go through.

The change appears safe, but let's add some documentation for future readers.

- Operator.h - Let's add examples of identity projections, e.g. all columns in Filter operator, group-by keys in aggregation. Let's also clarify that these are used to enable filter pushdown.

Reply:
Add comment of `identityProjections()`
```
  /// Returns a list of identity projections, e.g. columns that are projected
  /// as-is possibly after applying a filter. It is used to allow dynamic
  /// filters pushdown to go through. More examples of identity projections: all
  /// columns in FilterProject(only filters), group-by keys in HashAggregation.
```